### PR TITLE
Impl FromStr for MacAddr

### PR DESCRIPTION
**Pull request by clinedome**

### CODE REVIEW

**Comment by drunkirishcoder** [comment]
added a few things I needed on top of your PR and made a couple small adjustments. Main one is added an error struct specifically for `MacAddr`string parsing. `ParseError` is used for packet parsing errors (probably the name is not clear enough there).

File: framework/src/packets/ethernet.rs [hunk 1]
```diff
@@ -28,18 +30,25 @@ use std::fmt;
...
     pub fn new(a: u8, b: u8, c: u8, d: u8, e: u8, f: u8) -> Self {
         MacAddr([a, b, c, d, e, f])
     }
 
     pub fn new_from_slice(slice: &[u8]) -> Self {
         MacAddr([slice[0], slice[1], slice[2], slice[3], slice[4], slice[5]])
     }
+
+    /// Returns the six bytes the MAC address consists of
+    #[allow(clippy::trivially_copy_pass_by_ref)]
```
Comment (drunkirishcoder) [inline framework/src/packets/ethernet.rs]: had to turn this off. clippy is being annoying here. seems like a lint bug.

Comment (zeeshanlakhani) [reply framework/src/packets/ethernet.rs]: np. find to do that from time to time.

**Comment by zeeshanlakhani** [comment]
@drunkirishcoder lgtm on the added changes here.

**Pull request merged**